prov/efa/src/efa_domain.c

```diff
@@ -286,53 +286,38 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 
 	domain->cache = NULL;
 
-	/*
-	 * Check whether fork support is enabled when app does not request
-	 * FI_MR_LOCAL even if the cache is disabled.
-	 */
-	if (!app_mr_local && efa_check_fork_enabled(*domain_fid)) {
-		fprintf(stderr,
-		         "\nlibibverbs fork support is not supported by the EFA Libfabric\n"
-			 "provider when memory registrations are handled by the provider.\n"
-			 "\nFork support may currently be enabled via the RDMAV_FORK_SAFE\n"
-			 "or IBV_FORK_SAFE environment variable or another library in your\n"
-			 "application may be calling ibv_fork_init().\n"
-			 "\nPlease refer to https://github.com/ofiwg/libfabric/issues/6332\n"
-			 "for more information. Your job will now abort.\n");
-		abort();
-	}
-
 	/*
 	 * If FI_MR_LOCAL is set, we do not want to use the MR cache.
 	 */
-	if (!app_mr_local && efa_mr_cache_enable) {
-		domain->cache = (struct ofi_mr_cache *)calloc(1, sizeof(struct ofi_mr_cache));
-		if (!domain->cache) {
-			ret = -FI_ENOMEM;
-			goto err_free_info;
-		}
+	if (app_mr_local || !efa_mr_cache_enable)
```

Comment:
I think for normal code path to return early is a bit unusual. 

Would it make sense to put the whole section at the end into one function efa_mr_cache_init(), and do
```
if (!app_mr_local && efa_mr_cache_enable) {
     err = efa_mr_cache_init();
     if (err)
            efa_mr_cache_enable = false;
}

return 0;
```

Reply:
That is a much better idea, thanks for the suggestion.